apps/chat_server/lib/chat_server/contexts/rooms/update_room.ex

```diff
@@ -9,12 +10,13 @@ defmodule ChatServer.UpdateState do
     defstruct [:room]
   end
 
-  def call(params \\ %{}, _user, state) do
-    Logger.info "Updating room " <> inspect(params)
+  def call(room_id, _user, state) do
+    Logger.info "Updating room " <> inspect(room_id)
 
     # TODO verify user is allowed to update room of that type
 
-    with {:ok, record} <- update(params, state),
+    with record <- Schema.Room.get(room_id),
```

Comment:
Ecto `get` methods return the record struct instead of a record tuple, `{:ok, <record>}`. It makes sense, but there are times when a guarantee that a real record is returned and not nil would be better. Luckily, we can always write our own context to add that behaviour.

For that reason, it'd be better to use [`ChatServer.GetRoom`](https://github.com/oberon-chat/chat-server/blob/master/apps/chat_server/lib/chat_server/contexts/rooms/get_room.ex) instead:

```elixir
alias ChatServer.GetRoom

#...

with {:ok, record} <- GetRoom(room_id),
     {:ok, record} <- update(record, state),
```